libcst/matchers/_visitors.py

```diff
@@ -58,11 +65,11 @@ def _get_possible_match_classes(matcher: BaseMatcherNode) -> List[Type[cst.CSTNo
         return [getattr(cst, matcher.__class__.__name__)]
 
 
-def _get_possible_annotated_classes(annotation: Type[object]) -> List[Type[object]]:
```

Comment:
This needed a change since it is technically a lie. This function takes either a `Type[object]` or a `Union[Type[Object], ...]`. This was exposed by adding the above `TYPED_FUNCTION_RETURN_MAPPING`.